File: backend/Stage-Marsa-RDV-Management-backend/src/main/java/com/example/Rendez_vous_prise_container/Services/TrancheService.java
package com.example.Rendez_vous_prise_container.Services;

import com.example.Rendez_vous_prise_container.DTOs.TrancheAvailabilityDTO;
import com.example.Rendez_vous_prise_container.DTOs.TrancheRequestDTO;
import com.example.Rendez_vous_prise_container.Entities.Tranche;

import java.time.LocalDate;
import java.util.List;

public interface TrancheService {

    Tranche create(TrancheRequestDTO dto);

    Tranche update(Long id, TrancheRequestDTO dto);

    void delete(Long id);

    List<Tranche> getAll();

    Tranche getById(Long id);

    List<TrancheAvailabilityDTO> getAvailability(LocalDate date);
}

File: frontend/src/services/trancheservice.js
import api from './api';

export const trancheService = {
  getAll: () => api.get('/tranches'),
  getAvailability: (date) => api.get('/tranches/availability', { params: { date } }),
  getById: (id) => api.get(`/tranches/${id}`),
  create: (data) => api.post('/tranches', data),
  update: (id, data) => api.put(`/tranches/${id}`, data),
  delete: (id) => api.delete(`/tranches/${id}`),
};